didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+"""Quickstart helpers for beginner users of DIDComm."""
+from typing import Optional, Dict, List, Any, Union, Callable, Awaitable, Sequence
+import aiohttp
+import attr
+import attrs
+import json
+import logging
+import uuid
+
+from did_peer_2 import KeySpec, generate
+from pydid.did import DID
+from pydid import DIDDocument
+
+from aries_askar import Key, KeyAlg
+from didcomm_messaging import DIDCommMessaging
+from didcomm_messaging.crypto.backend.askar import AskarCryptoService, AskarSecretKey
+from didcomm_messaging.crypto.backend.basic import InMemorySecretsManager
+from didcomm_messaging.multiformats import multibase, multicodec
+from didcomm_messaging.packaging import PackagingService
+from didcomm_messaging.resolver import PrefixResolver
+from didcomm_messaging.resolver.peer import Peer2, Peer4
+from didcomm_messaging.routing import RoutingService
+
+JSON_OBJ = Dict[str, Any]
+Attachment = JSON_OBJ
+JSON_VALUE = Union[None, str, int, bool, float, JSON_OBJ, List[Any]]
+
+LOG = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class Message:
+    """Provide a nicer interface for messages than just Dictionaries."""
+
+    type: str
+    body: JSON_OBJ
+    id: Optional[str] = None
```

Comment:
Strong recommendation towards not using `attrs`; PyDID, a dependency of DMP is already using Pydantic so generally I would recommend using that instead OR not using a strict model class at all and just use a dataclass or something along those lines.

Reply:
I'll take a look. The primary reason I chose `attrs` was because it seemed simple and I didn't have to include an additional library that might be overly complicated.

Looking at the [documentation](https://docs.pydantic.dev/latest/), it seems simple enough so I'll look into switching to it.

Reply:
Due to discussion elsewhere, I have removed the message class